rcode-reader/src/main/java/eu/rssw/pct/RCodeInfo.java

```diff
@@ -0,0 +1,723 @@
+    private static final long serialVersionUID = 1L;
+
+    public InvalidRCodeException(String s) {
+      super(s);
+    }
+
+    public InvalidRCodeException(Throwable caught) {
+      super(caught);
+    }
+  }
+
+  private static Function parseProcSignature(String str) {
+    Function fn = new Function();
+    
+    // String is a comma-separated list of at least 3 elements
+    Iterator<String> lst =  Splitter.on(',').trimResults().split(str).iterator();
+    
+    // First element is the description, a space-separated list of 3 sub-elements
+    String desc = lst.next();
+    // Sub-element 1 is the type
+    fn.type = FunctionType.getTypeFromString(desc.substring(0, desc.indexOf(' ')));
+    // Sub-element 2 is the name (may contain spaces...)
+    fn.name = desc.substring(desc.indexOf(' ') + 1, desc.lastIndexOf(' '));
+    // Sub-element 3 is the flag
+    fn.accessTypes = SigAccessType.getTypeFromString(desc.substring(desc.lastIndexOf(' ') + 1));
+    
+    // Second element is return type, which may be empty (if entry doesn't return anything)
+    String retType = lst.next();
+    if (retType.isEmpty()) {
+      fn.returnType = DataType.VOID;
+    } else {
+      // Generics handling, which break comma-separated list...
+      if (retType.contains("[[")) {
+        while (!retType.contains("]]")) {
+          retType = retType + lst.next();
+        }
+      }
+      // If not empty, then two entries separated by a space, first is data type, then extent number
+      fn.returnExtent = Integer.parseInt(retType.substring(retType.lastIndexOf(' ') + 1));
+      fn.returnType = DataType.getDataType(retType.substring(0, retType.lastIndexOf(' ')));
+      if (fn.returnType == DataType.CLASS) {
+        fn.classReturnType = retType.substring(0, retType.indexOf(' '));
+      }
+    }
+    
+    // Parameters
+    while (lst.hasNext()) {
+      String str2 = lst.next();
+      if (!str2.isEmpty()) {
+        // Generics handling, which break comma-separated list...
+        if (str2.contains("[[")) {
+          while (!str2.contains("]]")) {
+            str2 = str2 + lst.next();
+          }
+        }
+
+        fn.parameters.add(parseParameter(str2));
+      }
+    }
+    
+    return fn;
+  }
+
+  private static Parameter parseParameter(String str) {
+    Parameter param = new Parameter();
+    List<String> prm = Splitter.on(' ').trimResults().splitToList(str);
+    if (prm.size() != 4) {
+      return param;
+    }
+    param.name = prm.get(1);
+    param.type = ParameterType.getTypeFromString(prm.get(0));
+    param.datatype = DataType.getDataType(prm.get(2));
+    param.extent = Integer.parseInt(prm.get(3));
+    param.classType = prm.get(2);
+
+    return param;
+  }
+
+  // Main object for all object signatures
+  public static class RCodeUnit {
+    private Function mainProcedure;
+    private Collection<Function> funcs = new ArrayList<>();
+    private List<TempTable> tts = new ArrayList<>();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "tts" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1068)